File: app/src/main/java/uz/mayasoft/marta/wallet/pin2/security/PFSecurityException.java
package uz.mayasoft.marta.wallet.pin2.security;


public class PFSecurityException extends Exception {

    private final Integer mCode;

    public PFSecurityException(String message, Integer code) {
        super(message);
        mCode = code;
    }

    public Integer getCode() {
        return mCode;
    }

    public PFSecurityError getError() {
        return new PFSecurityError(getMessage(), getCode());
    }
}

File: app/src/main/java/com/example/marta/ui/pin/security/PFSecurityException.kt
package com.example.marta.ui.pin.security

import com.example.marta.ui.pin.security.PFSecurityError

class PFSecurityException(
    message: String?,
    val code: Int
) :
    Exception(message) {
    val error: PFSecurityError
        get() = PFSecurityError(
            message!!,
            code
        )
}
